### PR TITLE
pkg/littlefs: bump littlefs version to 2.1.4

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The [littlefs](https://github.com/ARMmbed/littlefs) pkg hasn't been updated in a while.
There were small API Changes going from v1.6.2 -> v2.1.3

It's possible that littlefs got more efficient at allocating small files, but `statvfs()` would report no change when creating a 10 byte file - so I adjusted the unit test to create a bigger file.

### Testing procedure

I only checked that `tests/pkg_littlefs` still works on `native`.
I don't have a SPI flash file system set up right now.

### Issues/PRs references
none

### CODE REVIEW

**Comment by vincent-d** [comment]
We are using 1.x on the field and we have no way to migrate to 2.x. I'd like we find a way to keep both versions.

**Comment by benpicco** [comment]
Oh the on-disc format changed!
Well then this should be a new pkg - `littlefs2`.
Just have to figure out how to do this without lots of code duplication.

**Comment by vincent-d** [comment]
Maybe we could use pseudomodules to keep the glue code in the same place. But I don't know if we can deal with different values for `PKG_VERSION` to avoid duplicating this too.

**Comment by smlng** [comment]
Hi, can't we update the existing package to 1.7 (API should be unchanged). And then introduce littlefs2 as a separate package, and I would not try to have common glue code but simply copy what's needed even though we might have duplication. Because then we can remove littlefs in the future w/o problems.

**Comment by smlng** [comment]
so with #12664 merged this should be adapted to use `littlefs2` for package and module name, right?

File: pkg/littlefs/fs/littlefs_fs.c [hunk 1]
```diff
@@ -146,7 +146,13 @@ static int prepare(littlefs_desc_t *fs)
...
         fs->config.read_size = fs->dev->page_size;
     }
-    fs->config.lookahead = LITTLEFS_LOOKAHEAD_SIZE;
+    if (!fs->config.cache_size) {
+        fs->config.cache_size = fs->dev->page_size * LITTLEFS_CACHE_PAGES;
+    }
+    if (!fs->config.block_cycles) {
+        fs->config.block_cycles = LITTLEFS_BLOCK_CYCLES;
+    }
+    fs->config.lookahead_size = LITTLEFS_LOOKAHEAD_SIZE;
```
Comment (bergzand) [inline pkg/littlefs/fs/littlefs_fs.c]: [Previously](https://github.com/ARMmbed/littlefs/blob/v1.7.2/lfs.h#L155-L159), `config::lookahead` was defined as a number of bits. This was [changed](https://github.com/ARMmbed/littlefs/blob/v2.1.3/lfs.h#L209-L213) with the move to `config::lookahead_size` to bytes. This should be reflected in the allocated buffer [here](https://github.com/RIOT-OS/RIOT/blob/e8c78ad2ae88868770f21a00a92e13d17835365e/sys/include/fs/littlefs_fs.h#L100), the divide by 8 should be removed, or it should be added here.

With the current state of this PR, the memset [here](https://github.com/ARMmbed/littlefs/blob/v2.1.3/lfs.c#L3518) will overwrite large parts of the memory, exceeding the buffer limits.

**Comment by bergzand** [comment]
After out-of-band discussion with @benpicco, I've decided to open a separate PR and no longer make a mess out of this PR (see #13366) :)

@benpicco Feel free to remove my commits from this branch

**Comment by benpicco** [comment]
Thank you for taking over this PR!

**Pull request closed without merging**